### ISSUE TITLE
[Bug]: Web UI cannot fetch server data

### ISSUE THREAD

**Issue by MCgrant360** [spartacus04/jext-reborn#257]
While setting up the plugin, I reached the point of editing the config. I followed the directions regarding obtaining the link from the console. I put in my previously configured password, and it said "Error connecting to jext server" and "Failed to fetch" in the popup. I dried using my public ip, changing the port, forwarding the new port, and giving it firewall access, and none of these steps resolved the issue. I will upload the config as well in case it is a config issue on my end. 
[config.json](https://github.com/spartacus04/jext-reborn/files/14761943/config.json)

**Comment by spartacus04** [comment]
The config file is fine, however with this info I can't really find a problem.

Perhaps your ISP is blocking the port forwarding, for example mine shows the port forwarding settings on my router but doesn't actually allow them unless you buy the static IP option, but if you host your server from a data center that shouldn't be a problem.

Could you please try opening opening your browser devtools (usually by pressing ctrl-shift-I) and send me the network logs? You just need to go to the network tab, do everything as normal, export a .har file when it stops working.

**Comment by spartacus04** [comment]
Since .har files may contain personal data of the website session you can put them through [this](https://har-sanitizer.pages.dev/) sanitizer to remove it

**Comment by Quocnam201010** [comment]
Hello @spartacus04 
I am having the same issue

**Comment by spartacus04** [comment]
Since many people are affected by this problem, as a workaround while I try to fix it (and discover the issue) you can edit the `config.json` file in the plugin directory.

As for the discs you can just use the resource pack tool and download it to the plugin directory with the `resource-pack.zip` name.

**Comment by MCgrant360** [comment]
@spartacus04 your .HAR file is ready. This one is using a closed port, and no firewall access. If you want me to regenerate with a different configuration, please let me know.
[redacted_spartacus04.github.io.zip](https://github.com/spartacus04/jext-reborn/files/14774693/redacted_spartacus04.github.io.zip)

**Comment by spartacus04** [comment]
Apparently the connection was blocked by the generator itself and not so the request didn't even reach the plugin. Do the devtools console logs have any errors?

**Comment by spartacus04** [comment]
Apparently the connection was blocked by the generator itself and not so the request didn't even reach the plugin. Do the devtools console logs have any errors?

**Comment by spartacus04** [comment]
Apparently the connection was blocked by the generator itself and not so the request didn't even reach the plugin. Do the devtools console logs have any errors?

**Comment by MCgrant360** [comment]
There were errors in the console
![ConsoleErrors](https://github.com/spartacus04/jext-reborn/assets/120610323/e8037344-a4b7-4587-8a24-d8ac72eded16)

**Comment by MCgrant360** [comment]
@spartacus04 I have tried again. I attached the logs and .HAR file. Let me know if you need anything else.
[JEXT WEBUI logs.zip](https://github.com/spartacus04/jext-reborn/files/14891310/JEXT.WEBUI.logs.zip)

**Comment by Legendile7** [comment]
Having the same problem. Locally loading the resource-pack.zip is not working either. Is there any way to not use the web UI and make the plugin work? I am also using Geyser. I am on Paper 1.20.4

**Comment by Quocnam201010** [comment]
@spartacus04 I am having the same problem, people can't download self-hosted resource pack

**Comment by spartacus04** [comment]
> Also, how do I know the namespace of the discs? I really wish there was a /help and /listdiscs command. I don't understand why a web ui is being used and why the entire documentation is dependent on it. Just keep it stupid simple (KISS) like the original plugin.

I actually haven't removed the old disc loading with the new update, I should probably make it clearer in the quickstart and perhaps add a download button for the discs.json file only to the webui. It wasn't my intention to make the webui connection required, it was supposed to be a useful tool to people using the resource pack host.

About the commands you suggested as of now you can just use `/jext discs` and check the autocomplete, but i should probably make something like a /listdiscs

**Comment by spartacus04** [comment]
> @spartacus04 I have tried again. I attached the logs and .HAR file. Let me know if you need anything else. [JEXT WEBUI logs.zip](https://github.com/spartacus04/jext-reborn/files/14891310/JEXT.WEBUI.logs.zip)

could you try once, more? sorry about this situation, if you could also attach a screenshot of the console it would be perfect since the error logs do not contain every information.

**Comment by MCgrant360** [comment]
@spartacus04 The new logs are ready. I included a new .HAR, the jext-logs.txt and a screenshot of the dev console.
[JEXT-logsv2.zip](https://github.com/spartacus04/jext-reborn/files/14893894/JEXT-logsv2.zip)

**Comment by Legendile7** [comment]
> About the commands you suggested as of now you can just use `/jext discs` and check the autocomplete, but i should probably make something like a /listdiscs
There is no `/jext discs` command. There is `/jext disc` but the autocomplete on that puts in player names, not discs.

**Comment by spartacus04** [comment]
> > About the commands you suggested as of now you can just use `/jext discs` and check the autocomplete, but i should probably make something like a /listdiscs
> 
> There is no `/jext discs` command. There is `/jext disc` but the autocomplete on that puts in player names, not discs.

That's rather odd, the autocomplete works fine for me, did the plugin load fine? We should probably move this to a new issue btw

**Comment by spartacus04** [comment]
> @spartacus04 The new logs are ready. I included a new .HAR, the jext-logs.txt and a screenshot of the dev console. [JEXT-logsv2.zip](https://github.com/spartacus04/jext-reborn/files/14893894/JEXT-logsv2.zip)

There's something in the browser blocking the request, are you using any extensions/userscripts?

**Comment by MCgrant360** [comment]
> There's something in the browser blocking the request, are you using any extensions/userscripts?
@spartacus04 I use [UBlock Origin](https://chromewebstore.google.com/detail/ublock-origin/cjpalhdlnbpafiamejdnhcphjbkeiagm), [Microsoft Autofill](https://chromewebstore.google.com/detail/microsoft-autofill/fiedbfgcleddlbcmgdigjgdfcggjcion), and [Dark Theme](https://chromewebstore.google.com/detail/dark-theme-dark-reader-fo/eckokfcjbjbgjifpcbdmengnabecdakp). I can only imagine UBlock affecting web traffic.

**Comment by spartacus04** [comment]
> > There's something in the browser blocking the request, are you using any extensions/userscripts?
> @spartacus04 I use [UBlock Origin](https://chromewebstore.google.com/detail/ublock-origin/cjpalhdlnbpafiamejdnhcphjbkeiagm), [Microsoft Autofill](https://chromewebstore.google.com/detail/microsoft-autofill/fiedbfgcleddlbcmgdigjgdfcggjcion), and [Dark Theme](https://chromewebstore.google.com/detail/dark-theme-dark-reader-fo/eckokfcjbjbgjifpcbdmengnabecdakp). I can only imagine UBlock affecting web traffic.
> 
> 

That's strange, I use ublock too and it doesn't block the request. Try disabling it just to be sure though

**Comment by Legendile7** [comment]
That documentation update helped a lot, everything is working great now! Thanks!

**Comment by MCgrant360** [comment]
> That's strange, I use ublock too and it doesn't block the request. Try disabling it just to be sure though

@spartacus04 I disabled Ublock and it worked fine.
